lib/strings.nix

```diff
@@ -256,6 +256,28 @@ rec {
     # String to return if condition is true
     string: if cond then string else "";
 
+  /* Like `optionalString`, but returns `null` if `cond` is false.
```

Comment:
Why do we even need this function? Passing an empty string to build phases should be fine, no?

Reply:
@ElvishJerricco The intended meaning is the same as `optionalString` (use this value if this condition holds), it's purely a matter of different contexts having different no-ops (the empty string is the unit of string concatenation while null is the "unit" of derivation attribute assignment). Do you think `optionalString` is misnamed too?

@adamcstephens The reason is explained in the docstring: For derivations, adding an empty string attribute is not a no-op, it causes a rebuild. Adding a null is a no-op.

Reply:
The use case makes sense, but I also think this should have a better name. Maybe `stringOrNull` could be good.

Reply:
Also, we have [`lib` contribution guidelines](https://github.com/NixOS/nixpkgs/tree/master/lib#pr-guidelines) since recently, one point of which is to create separate PRs for each `lib` change, so this means splitting up this PR.

Reply:
OK, will split out the PR.

Reply:
Rebased on #285108